src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -295,7 +296,12 @@ private void RegisterLocalServices()
             this.localServices.Add(typeof(ISolutionRuleSetsInformationProvider), new Lazy<ILocalService>(() => new SolutionRuleSetsInformationProvider(this, Logger)));
             this.localServices.Add(typeof(IRuleSetSerializer), new Lazy<ILocalService>(() => new RuleSetSerializer(this)));
             this.localServices.Add(typeof(ISolutionBindingSerializer), new Lazy<ILocalService>(() => new SolutionBindingSerializer(this)));
-            this.localServices.Add(typeof(IConfigurationProvider), new Lazy<ILocalService>(() => new LegacyConfigurationProviderAdapter(this.GetService<ISolutionBindingSerializer>())));
+            this.localServices.Add(typeof(IConfigurationProvider), new Lazy<ILocalService>(() =>
+            {
+                var solution = this.GetService<SVsSolution, IVsSolution>();
```

Comment:
I decided to make the lazy service constructor do the work of fetching the _IVsSolution_. The alternative would have been to pass an _IServiceProvider_ to the _ConfigurationProvider_.